data_handler/handlers/loan_states/zklend/events.py

```diff
@@ -303,27 +305,39 @@ def process_borrowing_event(self, event: pd.Series) -> None:
                 )
             )
 
-    def process_repayment_event(self, event: pd.Series) -> None:
-        # The order of the values in the `data` column is: `repayer`, `beneficiary`, `token`, `raw_amount`,
-        # `face_amount`.
-        # Example: https://starkscan.co/event/0x06fa3dd6e12c9a66aeacd2eefa5a2ff2915dd1bb4207596de29bd0e8cdeeae66_5.
-        user = add_leading_zeros(event["data"][1])
-        token = add_leading_zeros(event["data"][2])
-
-        raw_amount = decimal.Decimal(str(int(event["data"][3], base=16)))
 
-        self.loan_entities[user].debt.increase_value(token=token, value=-raw_amount)
-        # add additional info block and timestamp
-        self.loan_entities[user].extra_info.block = event["block_number"]
-        self.loan_entities[user].extra_info.timestamp = event["timestamp"]
-        if user == self.verbose_user:
-            logging.info(
-                "In block number = {}, raw amount = {} of token = {} was repayed.".format(
-                    event["block_number"],
-                    raw_amount,
-                    token,
-                )
+def process_repayment_event(self, event: pd.Series) -> None:
+    # Parse event data into our RepaymentEventSerializer model to access by named attributes
+    parsed_event = RepaymentEventSerializer.parse_obj({
```

Comment:
insert only event, and parse it in `Serializer` not here.

Reply:
On it now

Reply:
@Akshola00 Ping me once you are done, I will be here 2-3 hours

Reply:
done